vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/SlideManagerTest.java

```diff
@@ -0,0 +1,102 @@
+import edu.asu.diging.vspace.core.model.ISlide;
+import edu.asu.diging.vspace.core.model.impl.Sequence;
+import edu.asu.diging.vspace.core.model.impl.Slide;
+
+public class SlideManagerTest {
+
+    @Mock
+    private SequenceRepository sequenceRepo;
+    
+    @Mock
+    private SlideRepository slideRepo;
+    
+    @InjectMocks
+    private SlideManager slideManagerToTest = new SlideManager();
+    
+    @InjectMocks
+    private SlideManager slideManagerToTestSpy =  Mockito.spy(slideManagerToTest);
+    
+    @Before
+    public void init() {
+        MockitoAnnotations.initMocks(this);
+    }
+    
+    // setting common used variables and Objects
+    String slideId, moduleId, sequenceId, slideIdNotInSequence;
+    List<ISlide> slidesList = Arrays.asList(new Slide());
+    Sequence sequenceObj;
+    
+    
+    @Before
+    public void setup() throws Exception {
+        sequenceObj = new Sequence() ;
+        slideId = "SLI000000002";
+        slideIdNotInSequence = "SLI000000219";
+        moduleId = "MOD000000002";
+        sequenceId = "SEQ000000004";
+        slidesList.get(0).setId(slideId);
+    }
+    
+    @Test
+    public void test_slideSequence() {
+        
+        // Positive scenario - Slide present in Sequence
+        
+        sequenceObj.setId(sequenceId);
+        sequenceObj.setSlides(slidesList);
+        List<Sequence> sequencesList = new ArrayList<>();
+        sequencesList.add(sequenceObj);
+        Mockito.when(sequenceRepo.findSequencesForModule(moduleId)).thenReturn(sequencesList);
+        List<Sequence> slideSequencePresent = slideManagerToTest.getSlideSequences(slideId, moduleId);
+        
+        if(slideSequencePresent.size() > 0) {
+            String actualSlideIdSequence = slideSequencePresent.get(0).getSlides().get(0).getId();
+            String expectedSlideIdSequence = sequencesList.get(0).getSlides().get(0).getId();
+            
+            Assert.assertEquals(actualSlideIdSequence, expectedSlideIdSequence);
+            Assert.assertEquals(slideSequencePresent.size(), sequencesList.size()); 
+        }
+        
+        // Negative Scenario - Slide not present in Sequence
+        List<Sequence> actualSequenceSlideListNotPresent = slideManagerToTest.getSlideSequences(slideId, moduleId);
+        Assert.assertNotSame(actualSequenceSlideListNotPresent, slideSequencePresent);
+    }
+    
+    @Test
+    public void test_deleteSlideById() throws SlideDoesNotExistException {
+        
+        
+        sequenceObj.setId(sequenceId);
+        sequenceObj.setSlides(slidesList);
+        List<Sequence> sequencesList = new ArrayList<>();
+        sequencesList.add(sequenceObj);
+        Mockito.when(sequenceRepo.findSequencesForModule(moduleId)).thenReturn(sequencesList);
+        
+        // using mockito spy object
+        ISlide slide = new Slide();
+        slide.setId(slideId);
+        Mockito.doReturn(slide).when(slideManagerToTestSpy).getSlide(Mockito.any());
```

Comment:
why do you need a spy object here?

Reply:
I am using spy object to mock the method of the same class. Otherwise, it's calling the actual method to get data which is null.

Reply:
you should do that on the manager that's being tested, you don't need a spy for that

Reply:
Yes, I am doing it on SlideManager that is being tested.
I tried to mock it normally, but it was still hitting DB and returning null. I checked on net, it suggested using Spy object, to mock method of the same class.

Reply:
you should mock the repository method that getSlide calls, so that get slides returns soemthing meaningful.